library/alloc/src/vec/mod.rs

```diff
@@ -2972,9 +2972,14 @@ impl<'a, T: Copy + 'a, A: Allocator + 'a> Extend<&'a T> for Vec<T, A> {
 
 /// Implements comparison of vectors, [lexicographically](Ord#lexicographical-comparison).
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T: PartialOrd, A: Allocator> PartialOrd for Vec<T, A> {
+impl<T, A1, A2> PartialOrd<Vec<T, A1>> for Vec<T, A2>
```

Comment:
Please make this `Vec<T, A1>` `partial_cmp` `Vec<T, A2>`, matching the existing PartialEq impl, instead of `Vec<T, A2>` `partial_cmp` `Vec<T, A1>`.

```suggestion
impl<T, A1, A2> PartialOrd<Vec<T, A2>> for Vec<T, A1>
```

![Screenshot from 2023-06-17 19-44-40](https://github.com/rust-lang/rust/assets/1940490/ca2ae067-3071-4b2d-bc4d-587246f08fe0)

Reply:
Updated, thanks!